thirdeye-ui/src/app/components/app-bar/app-bar-config-provider/app-bar-config-provider.component.tsx

```diff
@@ -74,11 +77,30 @@ export const AppBarConfigProvider: FunctionComponent<AppBarConfigProviderProps>
                         }
                     })
                 );
-                setWorkspace(workspaces![0]);
+                // Check if there is a workspace present in localStorage and whether its a valid
+                // workspace by checking it against fetched workspaces. If its present set it as
+                // workspace for current session else set the first workpsace from fetched data.
+                const sessionWorkspace = localPreferences.currentWorkspace;
+                const validSessionWorkspace = workspaces?.find((workspace) => {
```

Comment:
NIT.
```suggestion
                const isValidSessionWorkspace = workspaces?.find((workspace) => {
```

Reply:
the variables with is as prefix should be holding a boolean value I believe, and since here it holding the workspace, I didn't go with is as a prefix

Reply:
Makes sense.